book/src/csi-driver-object.md

```diff
@@ -0,0 +1,90 @@
+# CSI CRD: `CSIDriver` Object
+
+## What is the `CSIDriver` object?
+
+The `CSIDriver` Kubernetes API object serves two purposes:
+
+1. Ease driver discovery
+  * If a CSI driver creates a `CSIDriver` object, Kubernetes users can easily discover the CSI Drivers installed on their cluster (simply by issuing `kubectl get CSIDriver`)
+2. Customizing Kubernetes behavior
+  * Kubernetes has a default set of behaviors when dealing with CSI Drivers (for example, it calls the `Attach`/`Detach` operations by default). This object allows CSI drivers to specify how Kubernetes should interact with it.
+
+## What fields does the `CSIDriver` object have?
+
+Here is an example of a v1alpha1 `CSIDriver` object:
+
+```YAML
+apiVersion: csi.storage.k8s.io/v1alpha1
+kind: CSIDriver
+metadata:
+  name: mycsidriver.example.com
+spec:
+  attachRequired: true
+  podInfoOnMountVersion: v1
+```
+
+There are three important fields:
+
+* `name`
+  * This should correspond to the full name of the CSI driver.
+* `attachRequired`
+  * Indicates this CSI volume driver requires an attach operation (because it implements the CSI `ControllerPublishVolume` method), and that Kubernetes should call attach and wait for any attach operation to complete before proceeding to mounting.
+  * If a `CSIDriver` object does not exist for a given CSI Driver, the default is `true` -- meaning attach will be called.
+  * If a `CSIDriver` object exists for a given CSI Driver, but this field is not specified, it defaults to `false` -- meaning attach will not be called.
```

Comment:
If this field is not specified it defaults to `true`, i.e. attach will be called.

https://github.com/kubernetes/kubernetes/blob/18050e308c5603038947ec260ddd92e0c10e0002/pkg/volume/csi/csi_plugin.go#L626

Reply:
Good catch. Fixed